core/lib/sync/task.ts

```diff
@@ -31,19 +31,25 @@ function _sleep (time) {
 }
 
 function getUrl (task, currentToken, currentCartId) {
-  let url = task.url
-    .replace('{{token}}', (currentToken == null) ? '' : currentToken)
-    .replace('{{cartId}}', (currentCartId == null) ? '' : currentCartId)
+  const parsedUrl = queryString.parseUrl(task.url)
 
-  url = processURLAddress(url); // use relative url paths
-  if (config.storeViews.multistore) {
-    url = adjustMultistoreApiUrl(url)
+  if (parsedUrl.query.token && parsedUrl.query.token === '{{token}}') {
```

Comment:
I think you could use:
```js
if (parsedUrl.query && parsedUrl.query.token === '{{token}}') {
```

You could even go one step forward and use:
```ts
if (parsedUrl?.query?.token === '{{token}}') {
```

Reply:
Yes indeed, I've changed it.